src/Illuminate/Foundation/Application.php

```diff
@@ -245,6 +261,65 @@ protected function registerBaseServiceProviders()
         $this->register(new RoutingServiceProvider($this));
     }
 
+    /**
+     * Register the standard kernel classes for the application.
+     *
+     * @return $this
+     */
+    protected function withKernels()
+    {
+        $this->singleton(
+            \Illuminate\Contracts\Http\Kernel::class,
+            \App\Http\Kernel::class
+        );
+
+        $this->singleton(
+            \Illuminate\Contracts\Console\Kernel::class,
+            \App\Console\Kernel::class
+        );
+
+        return $this;
+    }
+
+    /**
+     * Register the standard exception handler for the application.
+     *
+     * @param  callable|null  $afterResolving
+     * @return $this
+     */
+    public function withExceptionHandling($afterResolving = null)
+    {
+        $this->singleton(
+            \Illuminate\Contracts\Debug\ExceptionHandler::class,
+            \Illuminate\Foundation\Exceptions\Handler::class
+        );
+
+        $this->afterResolving(
+            \Illuminate\Contracts\Debug\ExceptionHandler::class,
+            $afterResolving ?: fn ($handler) => $handler
+        );
+
+        return $this;
+    }
+
+    /**
+     * Register the braodcasting services for the application.
+     *
+     * @return $this
+     */
+    public function withBroadcasting()
+    {
+        $this->booted(function () {
+            Broadcast::routes();
+
+            if (file_exists($channels = $this->basePath('routes/channels.php'))) {
```

Comment:
having it as a config option would make it much more flexible

```php
if (isset($this->app->config['broadcasting']['paths'])) {
    // ...
}
```

Reply:
If I'm not mistaken, config isn't loaded at this point in the app lifecycle and only happens later on when the kernel is instantiated.

Reply:
Maybe an optional param of `withBroadcating` then? I admit I've never customized that filename, but hardcoding the app structure in the framework doesn't feel like the optimal approach :)